### PR TITLE
Adding parser for Stockholm files to Bio.Align

### PR DESCRIPTION
**Pull request by mdehoon**
Adding aligner for files in the Stockholm format to `Bio.Align`. Reading and writing alignments was tested on the complete Pfam, Rfam, and CATH databases.

- [X] I hereby agree to dual licence this and any previous contributions under both
the _Biopython License Agreement_ **AND** the _BSD 3-Clause License_.

- [X] I have read the ``CONTRIBUTING.rst`` file, have run ``pre-commit`` locally,
and understand that AppVeyor and TravisCI will be used to confirm the Biopython unit
tests and style checks pass with these changes.

- [X] I have added my name to the alphabetical contributors listings in the files
``NEWS.rst`` and ``CONTRIB.rst`` as part of this pull request, am listed
already, or do not wish to be listed. (*This acknowledgement is optional.*)

### CODE REVIEW

File: Tests/test_Align_stockholm.py [hunk 1]
```diff
@@ -63,274 +90,1573 @@ def test_reading_example(self):
...
- [0, 0, 1, 2, 6, 18, 18, 45, 46, 49, 50, 54, 77, 78, 79, 79, 145, 146, 146, 146],
- [0, 0, 1, 2, 6, 18, 18, 45, 46, 49, 50, 54, 77, 78, 79, 79, 145, 146, 146, 146],
- [0, 1, 1, 2, 6, 18, 18, 45, 46, 49, 50, 54, 77, 78, 79, 79, 145, 146, 146, 146],
- [0, 1, 1, 2, 6, 18, 18, 45, 46, 49, 50, 54, 77, 78, 79, 79, 145, 146, 146, 146],
- [0, 1, 1, 2, 6, 18, 18, 45, 46, 49, 50, 54, 77, 78, 79, 79, 145, 146, 146, 146],
- [0, 1, 1, 2, 6, 18, 18, 45, 46, 49, 50, 54, 77, 78, 79, 79, 145, 145, 146, 146],
- [0, 1, 1, 2, 6, 18, 18, 45, 46, 49, 50, 54, 77, 78, 79, 79, 145, 146, 146, 146],
- [0, 0, 1, 2, 6, 18, 18, 45, 46, 49, 50, 54, 77, 78, 79, 79, 145, 145, 145, 145],
-]
+                    [
```
Comment (peterjc) [inline Tests/test_Align_stockholm.py:115]: This strikes me as an example where we should locally disable black with:

```
# fmt: off
...
# fmt: on
```

Or if it works on multi-line statements, I saw recently they added ``# fmt: skip`` for use on single lines.

Comment (mdehoon) [reply Tests/test_Align_stockholm.py]: I have done that now, but `CI / style` seems to ignore `# fmt: off`, `# fmt: on`. It works when I run black locally.

Comment (peterjc) [reply Tests/test_Align_stockholm.py]: Right - black is not doing its stuff, but apparently there are still flake8 issues - especially loads of ``E241 multiple spaces after ','``

```
Tests/test_Align_stockholm.py:116:1: E122 continuation line missing indentation or outdented
Tests/test_Align_stockholm.py:161:1: E124 closing bracket does not match visual indentation
Tests/test_Align_stockholm.py:888:1: E122 continuation line missing indentation or outdented
Tests/test_Align_stockholm.py:896:1: E124 closing bracket does not match visual indentation
Tests/test_Align_stockholm.py:1059:1: E122 continuation line missing indentation or outdented
Tests/test_Align_stockholm.py:1059:3: E201 whitespace after '['
Tests/test_Align_stockholm.py:1059:7: E241 multiple spaces after ','
...
```

The indentation ought to be easy to fix, but you are appear to be deliberately breaking E241 in order to visually align columns of numbers. If you really want to do that, then I think each row of the matrix needs a ``  # noqa: E241`` comment :(

Comment (mdehoon) [reply Tests/test_Align_stockholm.py]: > but you are appear to be deliberately breaking E241 in order to visually align columns of numbers

That is how numpy prints it.

Comment (mdehoon) [reply Tests/test_Align_stockholm.py]: > If you really want to do that, then I think each row of the matrix needs a # noqa: E241 comment :(

I don't care enough about that. Can flake8 fix whatever it is complaining about?

**Comment by peterjc** [comment]
We should probably squash-and-merge this with an edited comment.

What was the Windows issue? Encodings?

**Comment by mdehoon** [comment]
> What was the Windows issue? Encodings?

Yes - `alignments = stockholm.AlignmentIterator(path)` would call `open` internally on `path`; the encoding then depends on the locale. I replaced it with
```python
        with open(path, encoding="UTF-8") as stream:
            # encoding depends on locale by default
            alignments = stockholm.AlignmentIterator(stream)
```
to specify the encoding explicitly.

**Pull request merged**